Few simple changes to allow code to cleanly compile on Linux/MacOS/Windows (VS2017/19) - add github action to do builds / add unit tests

alastairUK opened the pull request:
Some simple changes to automatically build on Window/Linux/Mac OS using GitHub actions. Have upped the warning level to W4 and fixed a few issues. Also added a very simple set of unit tests.

One thing my linter is picking up is the use of reserved identifiers (basically __)

https://clang.llvm.org/extra/clang-tidy/checks/bugprone-reserved-identifier.html

This is obviously all over the code as it's part of your coding standard. But something to be aware of I guess.

opsaunders reviewed (approved):
testing

wkozmaNTIA commented on src/p528/Troposcatter.cpp:
```diff
@@ -18,7 +17,7 @@
  |
  *===========================================================================*/
-void Troposcatter(Path *path, Terminal *terminal_1, Terminal *terminal_2, double d__km, double f__mhz, TroposcatterParams *tropo)
+void Troposcatter(Path * /*path*/, Terminal *terminal_1, Terminal *terminal_2, double d__km, double f__mhz, TroposcatterParams *tropo)
```
I'm not sure I understand what you are doing here?

wkozmaNTIA reviewed (approved):
I have 1 comment for you to respond to.  Also, I will probably merge this but remove the github actions file for the time being.  You can resubmit that as a new pull request, but not ready for that at the moment.  Want to update the Linux support and the couple other code quality items you suggest.

wkozmaNTIA commented:
Even though I approved this request, @alastairUK if you could respond to the above question that would be appreciated.  I'm holding off merging for a bit to give you a chance to respond

alastairUK commented:
I was compiling with these options

![image](https://user-images.githubusercontent.com/5337814/173389061-1c0cc5fd-04c8-4b72-a6dd-8b7fd35c0fe3.png)

So warning are errors.

Without that change you get...

![image](https://user-images.githubusercontent.com/5337814/173390470-fdbe6458-57c8-498f-8f6f-a8fbc67e3326.png)

I have rebased this PR onto cmake-support branch and fixed a couple more things

Note the github action compiles cleanly

https://github.com/alastairUK/p528/runs/6864839884

![image](https://user-images.githubusercontent.com/5337814/173394411-c5e3095e-ad8d-4f80-94f4-ed84df274d98.png)
